File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/DrawRndDigits.java
package Auxiliary_Modules;

import java.util.ArrayList;
import java.util.Random;

public class DrawRndDigits {
    // draw a random integer, then take said element out
    private ArrayList<Integer> strings = new ArrayList<>();
    private final ArrayList<Integer> stringsSource = new ArrayList<>();
    private final Random rand = new Random();
    public DrawRndDigits(int... values) {
        for (int value : values) {
            strings.add(value);
            stringsSource.add(value);
        }
    }
    public void addElement(int element){
        strings.add(element);
        stringsSource.add(element);
    }
    public int draw(){
        if (strings.isEmpty()) {return -1;}

        int x = rand.nextInt(strings.size());
        int element = strings.get(x);
        strings.remove(x);
        return element;
    }
    public int getSimpleRNDNum(int bound){
        // return 0->bound-1
        return rand.nextInt(bound);
    }
    public void reset(){
        DeepCopier dc = new DeepCopier();
        strings = dc.copyListOfInts(stringsSource);
    }
    public boolean isEmptied() {
        return strings.size() == 0;
    }

    public void resetIfEmpty() {
        if (strings.size() == 0) {
            reset();
        }
    }

    public boolean containsElement(int element) {
        return stringsSource.contains(element);
    }

    public boolean currentlyContainsElement(int element) {
        return strings.contains(element);
    }

    public void removeItem(int element) {
        strings.remove(element);
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/DrawRndDigits.kt
package auxiliary_modules

import java.util.*

class DrawRndDigits(vararg values: Int) {
    // draw a random integer, then take said element out
    private var strings = ArrayList<Int>()
    private val stringsSource = ArrayList<Int>()
    private val rand = Random()

    init {
        for (i in values.indices) {
            strings.add(values[i])
            stringsSource.add(values[i])
        }
    }

    fun addElement(element: Int) {
        strings.add(element)
        stringsSource.add(element)
    }

    fun draw(): Int {
        if (strings.isEmpty()) {
            return -1
        }
        val x = rand.nextInt(strings.size)
        val element = strings[x]
        strings.removeAt(x)
        return element
    }

    fun getSimpleRNDNum(bound: Int): Int {
        // return 0->bound-1
        return rand.nextInt(bound)
    }

    fun reset() {
        val dc = DeepCopier()
        strings = dc.copyListOfInts(stringsSource)
    }
    fun isEmptied(): Boolean {
        return strings.size == 0
    }

    fun resetIfEmpty() {
        if (strings.size == 0) {
            reset()
        }
    }

    fun containsElement(element: Int): Boolean {
        return stringsSource.contains(element)
    }

    fun currentlyContainsElement(element: Int): Boolean {
        return strings.contains(element)
    }

    fun removeItem(element: Int) {
            strings.removeAt(element)
    }
}